splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java

```diff
@@ -1398,6 +1406,7 @@ public void updateSystemSchemasView(TransactionController tc) throws StandardExc
         // we need to re-generate the metadataSPS due to the definition change of sysschemasview
         updateMetadataSPSes(tc);
 
+        tc.commit();
         SpliceLogUtils.info(LOG, "SYSVW.SYSSCHEMAVIEW updated to " + metadataRestrictionEnabled);
```

Comment:
Is this commit absolutely necessary?

Reply:
Not absolutely necessary. I put commit at the begin and end of this function so that the update of the schema view starts from a clean transaction and commit, all self-contained in this function. I'm not aware of side effect of redundant commit, if there is, we can remove.